src/main/java/com/monitorjbl/xlsx/StreamingReader.java

```diff
@@ -111,11 +127,57 @@ private void handleEvent(XMLEvent event) throws SAXException {
         currentCell = new StreamingCell(CellReference.convertColStringToIndex(coord[0]), Integer.parseInt(coord[1]) - 1);
 
         Attribute type = startElement.getAttributeByName(new QName("t"));
+/*ForgetMeNot--ibell--20150911--old code which didn't determine numeric vs date data types.
         if(type != null) {
           currentCell.setType(type.getValue());
         }
+*/
+        //
+        // This block will determine the value type and set the appropriate
+        // XSSF data type (used to format numeric cell values).
+        //
+        final String typeValue = (type != null) ? type.getValue() : "";
+        if ("b".equals(typeValue)) {
+          currentCell.setXssfDataType(StreamingCell.XSSF_DATA_TYPE.BOOLEAN);
+        } else if ("e".equals(typeValue)) {
+          currentCell.setXssfDataType(StreamingCell.XSSF_DATA_TYPE.ERROR);
+        } else if ("str".equals(typeValue)) {
+          currentCell.setXssfDataType(StreamingCell.XSSF_DATA_TYPE.FORMULA);
+        } else if ("inlineStr".equals(typeValue)) {
+          currentCell.setXssfDataType(StreamingCell.XSSF_DATA_TYPE.INLINE_STRING);
+        } else if ("s".equals(typeValue)) {
+          currentCell.setXssfDataType(StreamingCell.XSSF_DATA_TYPE.SST_INDEX);
+        } else {
```

Comment:
Copied solution from the references I gave in the pull request.  I believe that the type value could even be null for numeric (from my limited tests on date fields, specifically).  You would have to pore through the apache solution and/or source to determine if there is a specific numeric type.  I do not believe there is.  I believe it *is* the catchall.